lib/improver/tests/spotdata_new/spotdata_new/test_check_grid_match.py

```diff
@@ -0,0 +1,99 @@
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for check_grid_match function."""
+
+import unittest
+import numpy as np
+
+from iris.tests import IrisTest
+
+from improver.spotdata_new.spot_extraction import check_grid_match
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class Test_check_grid_match(IrisTest):
+
+    """Test the check_grid_match function."""
+
+    def setUp(self):
+        """Set up cubes for use in testing."""
+
+        attributes = {
+            'mosg__grid_domain': 'uk',
+            'mosg__grid_type': 'standard',
+            'mosg__grid_version': '1.2.0',
+            'mosg__model_configuration': 'uk_det'}
+
+        data = np.ones(9).reshape(3, 3).astype(np.float32)
+        self.reference_cube = set_up_variable_cube(data, attributes=attributes,
+                                                   spatial_grid="equalarea")
+        self.cube1 = self.reference_cube.copy()
+        self.cube2 = self.reference_cube.copy()
+
+    def test_matching_metadata(self):
+        """Test a case in which the grid metadata matches. There is no assert
+        statement as this test is successful if no exception is raised."""
+        cubes = [self.reference_cube, self.cube1, self.cube2]
+        check_grid_match('mosg', cubes)
+
+    def test_non_matching_metadata(self):
+        """Test a case in which the grid metadata does not match. This will
+        raise an ValueError."""
+        self.reference_cube.attributes["mosg__grid_domain"] = "eire"
+        cubes = [self.reference_cube, self.cube1, self.cube2]
+        msg = "Cubes do not share the metadata identified"
+        with self.assertRaisesRegex(ValueError, msg):
+            check_grid_match('mosg', cubes)
+
+    def test_ignore_non_matching_metadata(self):
+        """Test a case in which the grid metadata does not match but this is
+        forceably ignored by the user by setting self.grid_metadata_identifier
+        to None."""
+        self.reference_cube.attributes["mosg__grid_domain"] = "eire"
+        cubes = [self.reference_cube, self.cube1, self.cube2]
+        check_grid_match(None, cubes)
+
+    def test_no_identifier_success(self):
+        """Test case in which an empty string is provided as the identifier,
+        which matches all keys, assuming no numeric keys."""
+        cubes = [self.reference_cube, self.cube1, self.cube2]
+        check_grid_match('', cubes)
+
+    def test_no_identifier_failure(self):
```

Comment:
The `test_no_identifier_success` and `test_no_identifier_failure` seem to have exactly the same code, even though the docstring is different.